packages/smooth_app/lib/pages/product/common/product_list_popup_items.dart

```diff
@@ -216,6 +225,135 @@ class ProductListPopupOpenInWeb extends ProductListPopupItem {
   }
 }
 
+/// Popup menu item for the product list page: export list.
+class ProductListPopupExport extends ProductListPopupItem {
+  @override
+  String getTitle(final AppLocalizations appLocalizations) =>
+      appLocalizations.product_list_export;
+
+  @override
+  IconData getIconData() => Icons.download;
+
+  @override
+  ProductListPopupMenuEntry getEntry() => ProductListPopupMenuEntry.export;
+
+  @override
+  bool isDestructive() => false;
+
+  @override
+  Future<ProductList?> doSomething({
+    required final ProductList productList,
+    required final LocalDatabase localDatabase,
+    required final BuildContext context,
+  }) async {
+    final String listName = ProductQueryPageHelper.getProductListLabel(
```

Comment:
We may have a LOT of products in this list.
Using an `Isolate` would be far better.

Reply:
@g123k You may be too wary: it's just a String concatenation.
I would recommend to code the feature as simply as possible and to use the export feature normally: if that feature makes the app stutter in standard cases, of course we'll think about optimizations then. Including isolates.

Reply:
Honestly, this is just:
1) Create a `static` method
2) Call this method with `Isolate.run(() -> method())`

Reply:
Let's agree that we disagree :)

Reply:
The user actually clicks on a button to start the export, so she may expect something to be processing and be a little slow.
As opposed to where Isolates are used in the rest of the app: at the startup for an unexpected newsfeed retrieval, and during the image processing background tasks that aren't expected either by the user.
So, again, I suggest we put this idea of "isolate" on the shelf and see later how much users complain about export performances.